lib/md051.js

```diff
@@ -6,7 +6,8 @@ const { addError, escapeForRegExp, filterTokens, forEachInlineChild,
   forEachHeading, htmlElementRe } = require("../helpers");
 
 // Regular expression for identifying HTML anchor names
-const identifierRe = /(?:id|name)\s*=\s*['"]?([^'"\s>]+)/iu;
```

Comment:
What about leaving it as a single RegExp, making id/name a capture, and looking at that?

Reply:
That would only capture one or the other, but it is valid to have both. For example, with `<input name=foo id=bar>`, the anchor should be `bar`, not `foo`. I could make it a global regexp, but I that would complicate the code (having to loop). I thought the two regexps, though more verbose, was a bit easier to follow.

But if you disagree, let me know and I'll give it a shot.

(I just added a test case to cover such a potential regression.)

Reply:
I hadn't thought of both being present. What you've done is good!